docs/examples/301/servicebus-namespace-vnet/main.bicep

```diff
@@ -0,0 +1,46 @@
+param serviceBusNamespaceName string
+param virtualNetworkName string
+param subnetName string
+param location string = resourceGroup().location
+
+resource serviceBusNamespace 'Microsoft.ServiceBus/namespaces@2018-01-01-preview' = {
+  name: serviceBusNamespaceName
+  location: location
+  sku: {
+    name: 'Premium'
+    tier: 'Premium'
+  }
+  properties: {}
+}
+
+resource virtualNetwork 'Microsoft.Network/virtualNetworks@2017-09-01' = {
+  name: virtualNetworkName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        '10.0.0.0/23'
+      ]
+    }
+    subnets: [
+      {
+        name: subnetName
+        properties: {
+          addressPrefix: '10.0.0.0/23'
+          serviceEndpoints: [
+            {
+              service: 'Microsoft.ServiceBus'
+            }
+          ]
+        }
+      }
+    ]
+  }
+}
+
+resource namespaceVirtualNetworkRule 'Microsoft.ServiceBus/namespaces/VirtualNetworkRules@2018-01-01-preview' = {
```

Comment:
I think this would be a little clearer to understand with the following:
```bicep
resource namespaceVirtualNetworkRule 'Microsoft.ServiceBus/namespaces/VirtualNetworkRules@2018-01-01-preview' = {
  name: '${serviceBusNamespace.name}/${virtualNetworkName}'
  properties: {
    virtualNetworkSubnetId: '${virtualNetwork.id}/subnets/${subnetName}'
  }
}
```

My reasoning:
1. It's just by convention that the name matches the virtualNetwork's name, so feels like it is more understandable to use the variable `virtualNetworkname`.
1. I feel like it's a good practice to avoid using `resourceId()` unless absolutely necessary, just because it's easy to mess up with refactors/renames, and this way, the dependency is implicit.